src/plugins/external_tool_support/src/ExternalToolManager.cpp

```diff
@@ -343,7 +348,10 @@ void ExternalToolManagerImpl::sl_customToolsLoaded(Task *task) {
     StrStrMap toolPaths;
     foreach (ExternalTool *tool, toolsList) {
         SAFE_POINT(tool, "Tool is NULL", );
-        QString toolPath = addTool(tool);
+        addTool(tool);
+    }
+    foreach (ExternalTool *tool, toolsList) {
+        QString toolPath = addToolToListsAndReturnToolPath(*tool);
```

Comment:
Here you do not have the SAFE_POINT like in the loop above.

Reply:
1. If `toolsList` contains `nullptr`, `SAFE_POINT` will be triggered and we will return from function
2.` toolsList` does not change in above loop
3. Then `nullptr` cannot be inserted in the `toolsList`
4. This is why `SAFE_POINT` is redundant in the second loop

Reply:
toolsList is obtained from etRegistry->getAllEntries() which must never have nulls.
This is the reason why I asked about unnecessary safepoints in the code above.